presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -1021,4 +1050,54 @@ default MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession
     {
         return new MaterializedViewFreshness(false);
     }
+
+    /**
+     * Redirects to another table or view which may or may not in the same catalog.
+     * <p>
+     * This method is called whenever the engine references a table or view name, including both accessing and creating one. Connectors are
+     * expected to return {@link Optional#empty()} if the redirection doesn't happen. The engine calls this method in an iterative manner,
+     * i.e., using the redirected name as the argument for the next call, until an {@link Optional#empty()} is returned. If a redirection
+     * does happen, the engine will access the redirected table or view name instead of the original one and raise a REDIRECTED_TABLE warning.
+     * </p>
+     * <p>
+     * Here are some caveats worth noting:
+     * <li>
+     *     If you choose to implement this method, you should consider implementing {@link #listTableColumnsWithRedirection}, {@link #getViewsWithRedirection},
+     *     {@link #listTablePrivilegesWithRedirection}, too. These methods are called to answer information schema queries. You should also
+     *     consider implementing {@link #redirectTableRename}, which is used to redirect rename targets.
+     * </li>
+     * <li>
+     *     Be careful if you implement redirection to another catalog. Different catalogs have different sets of roles. In other words, GRANT/REVOKE
+     *     queries may break due to the non-existence of certain roles in the new catalog. Different connectors may have different syntax, feature sets,
+     *     type support, etc., too. So some working queries with one connector may break with another connector. You should make sure your production
+     *     queries are not among them.
```

Comment:
That's odd to mention "production queries" here.

This comment is for SPI implementor. From SPI perspective there is no concept of "production queries" or "production" at all. From SPI perspective, breaking any queries is equally bad, not just those regarded as "production".

Actually, SPI implementor and a person responsible for a production environment are not necessarily the same person, so "you" reference here is ambiguous.

Please remove this part.

Actually, all those considerations should be part of `hive.redirect-to-iceberg-enabled` documentation.